vllm/model_executor/layers/sampler.py

```diff
@@ -490,6 +508,9 @@ def _get_logprobs(
         batched_logprobs_query_seq_indices,
         batched_logprobs_query_token_indices
     ]]
+    batched_ranks_query_result = _get_ranks(
+        logprobs[batched_logprobs_query_seq_indices],
+        batched_logprobs_query_token_indices)
```

Comment:
let's move this after line 526, since we'll have to force a CPU<->GPU sync here anyway.

Reply:
will do!